scipy/optimize/zeros.py

```diff
@@ -648,3 +667,450 @@ def brenth(f, a, b, args=(),
+        if fc == 0:
+            return self.get_result(c)
+
+        self.d, self.fd = self._update_bracket(c, fc)
+        self.e, self.fe = None, None
+        self.iterations += 1
+
+        while True:
+            status, xn = self.iterate()
+            if status == _ECONVERGED:
+                return self.get_result(xn)
+            if status == _ECONVERR:
+                if disp:
+                    msg = "Failed to converge after %d iterations, bracket is %s" % (self.iterations + 1, self.ab)
+                    raise RuntimeError(msg)
+                return self.get_result(xn, _ECONVERR)
+
+
+def toms748(f, a, b, args=(), k=2,
+            xtol=_xtol, rtol=_rtol, maxiter=_iter,
+            full_output=False, disp=True):
+    """
+    Find a zero using TOMS Algorithm 748 method.
+
+    Implements the Algorithm 748 method of Alefeld, Potro and Shi to find a
+    zero of the function `f` on the sign changing interval `[a , b]`.
+    It uses a mixture of inverse cubic interpolation with
+    "Newton-quadratic" steps. [APS1995]
+    The R-order is at least 2.7, and with about asymptotically 2 function
+    evaluation per iteration, the efficiency index is approximately 1.65.
+
+    Parameters
+    ----------
+    f : function
+        Python function returning a number.  The function :math:`f`
+        must be continuous, and :math:`f(a)` and :math:`f(b)`
+        have opposite signs.
+    a : number
+    b : number, with b > a and f(b)*f(a) < 0
+    args : tuple, optional
+        containing extra arguments for the function `f`.
+        `f` is called by ``f(x, *args)``.
+    xtol : number, optional
+        The computed root ``x0`` will satisfy ``np.allclose(x, x0,
+        atol=xtol, rtol=rtol)``, where ``x`` is the exact root. The
+        parameter must be nonnegative.
+    rtol : number, optional
+        The computed root ``x0`` will satisfy ``np.allclose(x, x0,
+        atol=xtol, rtol=rtol)``, where ``x`` is the exact root.
+    maxiter : number, optional
+        if convergence is not achieved in maxiter iterations, an error is
+        raised.  Must be >= 0.
+    full_output : bool, optional
+        If `full_output` is False, the root is returned.  If `full_output` is
+        True, the return value is ``(x, r)``, where `x` is the root, and `r` is
+        a RootResults object.
+    disp : bool, optional
+        If True, raise RuntimeError if the algorithm didn't converge.
+        Otherwise the convergence status is recorded in the RootResults
+        return object.
+
+    Returns
+    -------
+    x0 : float
+        Approximate Zero of `f`
+    r : RootResults (present if ``full_output = True``)
+        Object containing information about the convergence.  In particular,
+        ``r.converged`` is True if the routine converged.
+
+    See Also
+    --------
+    brentq, brenth, ridder, bisect, newton
+    fsolve : find zeroes in n dimensions.
+
+    Notes
+    -----
+    `f` must be continuous.
```

Comment:
Could you say something here on when this solver is preferred / more performant than the other ones we have?